apps/backend/src/collectivites/discussions/domain/discussion-domain-service.spec.ts

```diff
@@ -0,0 +1,369 @@
+import { Logger } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { beforeEach, describe, expect, test, vi } from 'vitest';
+
+import { DiscussionMessageRepository } from '@/backend/collectivites/discussions/infrastructure/discussion-message-repository.interface';
+import { DiscussionRepository } from '@/backend/collectivites/discussions/infrastructure/discussion-repository.interface';
+import { DiscussionDomainService } from './discussion-domain-service';
+import {
+  CreateDiscussionData,
+  DiscussionErrorEnum,
+  DiscussionStatutEnum,
+  DiscussionType,
+  ReferentielEnum,
+} from './discussion.type';
+
+describe('DiscussionDomainService', () => {
+  let service: DiscussionDomainService;
+  let mockDiscussionRepository: DiscussionRepository;
+  let mockDiscussionMessageRepository: DiscussionMessageRepository;
+  let mockLogger: Logger;
+
+  const mockDiscussion: DiscussionType = {
+    id: 1,
+    collectiviteId: 123,
+    actionId: 'action-1',
+    status: DiscussionStatutEnum.OUVERT,
+    createdBy: 'user-id-1',
+    createdAt: '2025-01-01T00:00:00.000Z',
+    modifiedAt: '2025-01-01T00:00:00.000Z',
+  };
+
+  const mockMessage = {
+    id: 1,
+    discussionId: 1,
+    message: 'Test message',
+    createdBy: 'user-id-1',
+    createdByNom: 'Test User',
+    createdAt: '2025-01-01T00:00:00.000Z',
+  };
+
+  beforeEach(async () => {
+    mockDiscussionRepository = {
+      findById: vi.fn(),
+      findOrCreate: vi.fn(),
+      list: vi.fn(),
+      update: vi.fn(),
+    } as unknown as DiscussionRepository;
+
+    mockDiscussionMessageRepository = {
+      create: vi.fn(),
+      delete: vi.fn(),
+      findByDiscussionIds: vi.fn(),
+    } as unknown as DiscussionMessageRepository;
+
+    mockLogger = {
+      log: vi.fn(),
+      error: vi.fn(),
+      warn: vi.fn(),
+      debug: vi.fn(),
+      verbose: vi.fn(),
+    } as unknown as Logger;
+
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [
+        DiscussionDomainService,
+        {
+          provide: 'DiscussionRepository',
+          useValue: mockDiscussionRepository,
```

Comment:
et ici 
```suggestion
          useValue: mockDiscussionRepository as DiscussionRepository,
```

De cette façon TS est efficace même dans les tests